src/content/tools/hooks.md

```diff
@@ -0,0 +1,377 @@
+packages.
+:::
+
+Example dependencies for a build hook:
+
+```yaml title="pubspec.yaml"
+name: native_add_library
+description: Sums two numbers with native code.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.8.0 <4.0.0'
+
+dependencies:
+  ... 
+  code_assets: any
+  hooks: any
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+ Example dependencies for a link hook:
+
+ ```yaml title="pubspec.yaml"
+name: package_with_assets
+description: An example of using a package with assets.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.8.0 <4.0.0'
+
+dependencies:
+  ...
+  data_assets: any
+  hooks: any
+
+dev_dependencies:
+  ...
+ ```
+
+### Create a build hook script
+
+If you want to use a build hook, first create the native
+library that you want to use, and then follow these steps to
+connect a build hook to that library in a Dart script: 
+
+1.  In your Dart project, create or open `hooks/build.dart`. 
+
+1.  In the main method, add a `build` function from
+    `package:hooks/hooks.dart` and link to the native
+    library. For example:
+
+    ```dart title="hooks/build.dart" highlightLines=6
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    * `input`: The input for `hook/build.dart`. This
+      includes the target OS, target architecture,
+      output directory, etc. For details, see
+      the [`BuildInput`][] class.
+
+    * `output`: The output for `hook/build.dart`. This
+      includes the assets to be bundled, and dependencies.
+      For details, see the [`BuildOutputBuilder`][] class.
+
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+
+### Create a link hook script
+
+If you want to use a link hook, follow these steps to
+add one to a Dart script: 
+
+1.  In your Dart project, create or open `hooks/link.dart`. 
+
+1.  In the main method, add a `link` function from
+    `package:hooks/hooks.dart` and construct a data asset.
+    For example:
+
+    ```dart title="hooks/link.dart" highlightLines=14
+    import 'dart:convert';
+    import 'dart:io';
+
+    import 'package:data_assets/data_assets.dart';
+    import 'package:hooks/hooks.dart';
+    import 'package:record_use/record_use.dart';
```

Comment:
Record use is very experimental, we shouldn't show this yet.

Note that without record use you can't really use a link hook to shrink assets
yet. So maybe it would be better to leave link hooks out completely for now.